### PR TITLE
[Do not merge] Just share my config changes

### PR DESCRIPTION
**Pull request by hebiao064**
Result looks good
image

### CODE REVIEW

**Review by cedricbeta — APPROVED** [review]

**Comment by hebiao064** [comment]
(WorkerDict pid=2297617)   warnings.warn(
Online DPO Training Progress:   3%|▎         | 3/87 [18:17<8:10:37, 350.44s/it, global_seqlen/min=66165, global_seqlen/max=71649, global_seqlen/minmax_diff=5484, global_seqlen/balanced_min=68768, global_seqlen/balanced_max=68768, global_seqlen/mean=68768.000, time/validation_run=141.534, val-core/DigitalLearningGmbH/MATH-lighteval/reward/mean@1=0.508, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/std@1=0.000, val-core/DigitalLearningGmbH/MATH-lighteval/reward/best@1/mean=0.508, val-core/DigitalLearningGmbH/MATH-lighteval/reward/best@1/std=0.000, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/worst@1/mean=0.508, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/worst@1/std=0.000, reward/rewards/mean=0.557, reward/rewards/max=1.000, reward/rewards/min=0.000, response_length/mean=429.910, response_length/max=1024.000, response_length/min=61.000, response_length/clip_ratio=0.076, prompt_length/mean=2048.000, prompt_length/max=2048.000, prompt_length/min=2048.000, prompt_length/clip_ratio=1.000, timing_s/gen=13.073, timing_s/policy_log_prob=12.353, timing_s/ref_log_prob_dpo=12.548, timing_s/reward_calc=0.108, timing_s/prepare_dpo_batch=0.008, timing_s/update_actor=42.587, timing_s/step=428.743, timing_per_token_ms/update_actor=0.155, timing_per_token_ms/gen=0.059, perf/total_num_tokens=275072, perf/time_per_step=428.743, perf/throughput=160.395, time/steOnline DPO Training Progress:   5%|▍         | 4/87 [18:17<8:54:14, 386.20s/it, global_seqlen/min=66165, global_seqlen/max=71649, global_seqlen/minmax_diff=5484, global_seqlen/balanced_min=68768, global_seqlen/balanced_max=68768, global_seqlen/mean=68768.000, time/validation_run=141.534, val-core/DigitalLearningGmbH/MATH-lighteval/reward/mean@1=0.508, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/std@1=0.000, val-core/DigitalLearningGmbH/MATH-lighteval/reward/best@1/mean=0.508, val-core/DigitalLearningGmbH/MATH-lighteval/reward/best@1/std=0.000, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/worst@1/mean=0.508, val-aux/DigitalLearningGmbH/MATH-lighteval/reward/worst@1/std=0.000, reward/rewards/mean=0.557, reward/rewards/max=1.000, reward/rewards/min=0.000, response_length/mean=429.910, response_length/max=1024.000, response_length/min=61.000, response_length/clip_ratio=0.076, prompt_length/mean=2048.000, prompt_length/max=2048.000, prompt_length/min=2048.000, prompt_length/clip_ratio=1.000, timing_s/gen=13.073, timing_s/policy_log_prob=12.353, timing_s/ref_log_prob_dpo=12.548, timing_s/reward_calc=0.108, timing_s/prepare_dpo_batch=0.008, timing_s/update_actor=42.587, timing_s/step=428.743, timing_per_token_ms/update_actor=0.155, timing_per_token_ms/gen=0.059, perf/total_num_tokens=275072, perf/time_per_step=428.743, perf/throughput=160.395, time/step=428.744]
(WorkerDict pid=2297615) /home/jobuser/sglang/venv/lib/python3.10/site-packages/torch/distributed/fsdp/fully_sharded_data_parallel.py:690: FutureWarning: FSDP.state_dict_type() and FSDP.set_state_dict_type() are being deprecated. Please use APIs, get_state_dict() and set_state_dict(), which can support different parallelisms, FSDP1, FSDP2, DDP. API doc: https://pytorch.org/docs/stable/distributed.checkpoint.html#torch.distributed.checkpoint.state_dict.get_state_dict .Tutorial: https://pytorch.org/tutorials/recipes/distributed_checkpoint_recipe.html . [repeated 2x across cluster]
(WorkerDict pid=2297615)   warnings.warn( [repeated 2x across cluster]
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'chosen_logps' not found.
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'rejected_logps' not found.
(TaskRunner pid=2279888) ---- [DEBUG] Calculated DPO Data Metrics: ['reward/rewards/mean', 'reward/rewards/max', 'reward/rewards/min', 'response_length/mean', 'response_length/max', 'response_length/min', 'response_length/clip_ratio', 'prompt_length/mean', 'prompt_length/max', 'prompt_length/min', 'prompt_length/clip_ratio'] ----
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) [Step 3, Batch 2] Processing batch size: 256
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) [Step 3] Updating Reference Model Weights from Actor...
(WorkerDict pid=2297616) [rank-2]: Saving model to /tmp/actor_state_mid/model_world_size_4_rank_2.pt
(WorkerDict pid=2297616) [rank-2]: Saving checkpoint to /tmp/actor_state_mid/model_world_size_4_rank_2.pt
(WorkerDict pid=2297616) [rank-2]: Saving extra_state to /tmp/actor_state_mid/extra_state_world_size_4_rank_2.pt
(WorkerDict pid=2281491) [rank-0]: Loading from /tmp/actor_state_mid/model_world_size_4_rank_0.pt and /tmp/actor_state_mid/optim_world_size_4_rank_0.pt and /tmp/actor_state_mid/extra_state_world_size_4_rank_0.pt
(WorkerDict pid=2297617) [rank-3]: Saving model to /tmp/actor_state_mid/model_world_size_4_rank_3.pt [repeated 3x across cluster]
(WorkerDict pid=2297617) [rank-3]: Saving checkpoint to /tmp/actor_state_mid/model_world_size_4_rank_3.pt [repeated 3x across cluster]
(WorkerDict pid=2297617) [rank-3]: Saving extra_state to /tmp/actor_state_mid/extra_state_world_size_4_rank_3.pt [repeated 3x across cluster]
(TaskRunner pid=2279888) [Step 3] Reference Model Weights Updated.
(WorkerDict pid=2297617) [rank-3]: Loading from /tmp/actor_state_mid/model_world_size_4_rank_3.pt and /tmp/actor_state_mid/optim_world_size_4_rank_3.pt and /tmp/actor_state_mid/extra_state_world_size_4_rank_3.pt [repeated 3x across cluster]
(TaskRunner pid=2279888) ---- [Step 3] DEBUG DPO: Calling compute_ref_log_prob ----
(TaskRunner pid=2279888) ---- [Step 3] DEBUG DPO: ref_log_prob tensor shape: torch.Size([512, 1024]) ----
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) ---- [DEBUG] Entering compute_onlineDPO_pref ----
(TaskRunner pid=2279888)   Input batch keys: ['input_ids', 'attention_mask', 'prompts', 'responses', 'position_ids', 'response_mask', 'old_log_probs', 'ref_log_prob', 'token_level_rewards']
(TaskRunner pid=2279888)   Input token_level_rewards shape: torch.Size([512, 1024]), dtype: torch.float32
(TaskRunner pid=2279888)   Input token_level_rewards Min=0.0000, Max=1.0000, Mean=0.0005
(TaskRunner pid=2279888)   Input response_mask shape: torch.Size([512, 1024]), dtype: torch.int64
(TaskRunner pid=2279888)   Input response_mask Min=0, Max=1
(TaskRunner pid=2279888) ---- [DEBUG] Inside compute_onlinedpo_pref ----
(TaskRunner pid=2279888)   Calculated sequence scores shape: torch.Size([512])
(TaskRunner pid=2279888)   Reshaped score pairs shape: torch.Size([256, 2])
(TaskRunner pid=2279888)   Winner indices shape: torch.Size([256])
(TaskRunner pid=2279888)   Number where Response 2 (index 1) is preferred: 51
(TaskRunner pid=2279888)   Output preference mask shape: torch.Size([512])
(TaskRunner pid=2279888)   Output mask True count (Chosen): 256
(TaskRunner pid=2279888)   Output mask False count (Rejected): 256
(TaskRunner pid=2279888) ---- [DEBUG] Exiting compute_onlinedpo_pref ----
(TaskRunner pid=2279888) ---- [DEBUG] compute_onlineDPO_pref - Output ----
(TaskRunner pid=2279888)   Output preferences shape: torch.Size([512]), dtype: torch.bool
(TaskRunner pid=2279888)   Output preferences True count: 256, False count: 256
(TaskRunner pid=2279888) ---- [DEBUG] Exiting compute_onlineDPO_pref ----
(TaskRunner pid=2279888) WARNING: Creating DPO labels using configured max_prompt_length...
(TaskRunner pid=2279888) ---- [Step 3] DEBUG DPO: Prepared DPO Update Batch ----
(TaskRunner pid=2279888)   Keys: ['chosen_input_ids', 'chosen_attention_mask', 'chosen_labels', 'rejected_input_ids', 'rejected_attention_mask', 'rejected_labels', 'reference_chosen_logps', 'reference_rejected_logps', 'chosen_position_ids', 'rejected_position_ids']
(TaskRunner pid=2279888)   Meta Info: {'dpo_beta': 0.1, 'dpo_loss_type': 'sigmoid', 'dpo_label_smoothing': 0.0, 'use_reference_policy': True, 'reference_free': False, 'global_step': 3}
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) Running DPO validation at step 3...
(TaskRunner pid=2279888) test_gen_batch meta info: {'eos_token_id': 151645, 'pad_token_id': 151643, 'recompute_log_prob': False, 'do_sample': False, 'validate': True}
(TaskRunner pid=2279888) validation generation end
(TaskRunner pid=2279888) [prompt] system
(TaskRunner pid=2279888) You are a helpful assistant.
(TaskRunner pid=2279888) user
(TaskRunner pid=2279888) How many vertical asymptotes does the graph of $y=\frac{2}{x^2+x-6}$ have? Let's think step by step and output the final answer within \boxed{}.
(TaskRunner pid=2279888) assistant
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) [response] To find the vertical asymptotes of the function \(y = \frac{2}{x^2 + x - 6}\), we need to determine the values of \(x\) for which the denominator becomes zero, as these are the points where the function is undefined and potentially has vertical asymptotes.
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) First, let's factor the denominator:
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) \[x^2 + x - 6 = (x + 3)(x - 2)\]
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) Setting the denominator equal to zero gives us the equation:
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) \[(x + 3)(x - 2) = 0\]
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) Solving for \(x\) yields:
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) \[x + 3 = 0 \Rightarrow x = -3\]
(TaskRunner pid=2279888) \[x - 2 = 0 \Rightarrow x = 2\]
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) Thus, the function has vertical asymptotes at \(x = -3\) and \(x = 2\).
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) Therefore, the graph of \(y = \frac{2}{x^2 + x - 6}\) has **2** vertical asymptotes. The answer is:
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) \[
(TaskRunner pid=2279888) \boxed{2}
(TaskRunner pid=2279888) \]
(TaskRunner pid=2279888) [ground_truth] 2
(TaskRunner pid=2279888) [score] 1.0
(TaskRunner pid=2279888) ---- [DEBUG] Computing DPO Data Metrics ----
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'token_level_scores' not found.
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'dpo_logits' not found.
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'chosen_logps' not found.
(TaskRunner pid=2279888) DEBUG compute_dpo_data_metrics: 'rejected_logps' not found.
(TaskRunner pid=2279888) ---- [DEBUG] Calculated DPO Data Metrics: ['reward/rewards/mean', 'reward/rewards/max', 'reward/rewards/min', 'response_length/mean', 'response_length/max', 'response_length/min', 'response_length/clip_ratio', 'prompt_length/mean', 'prompt_length/max', 'prompt_length/min', 'prompt_length/clip_ratio'] ----
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) [Step 4, Batch 3] Processing batch size: 256
(TaskRunner pid=2279888) 
(TaskRunner pid=2279888) [Step 4] Updating Reference Model Weights from Actor...
(WorkerDict pid=2297617) /home/jobuser/sglang/venv/lib/python3.10/site-packages/torch/distributed/fsdp/fully_sharded_data_parallel.py:690: FutureWarning: FSDP.state_dict_type() and FSDP.set_state_dict_type() are being deprecated. Please use APIs, get_state_dict() and set_state_dict(), which can support different parallelisms, FSDP1, FSDP2, DDP. API doc: https://pytorch.org/docs/stable/distributed.checkpoint.html#torch.distributed.checkpoint.state_dict.get_state_dict .Tutorial: https://pytorch.org/tutorials/recipes/distributed_checkpoint_recipe.html .
(WorkerDict pid=2297617)   warnings.warn(
(WorkerDict pid=2297616) /home/jobuser/sglang/venv/lib/python3.10/site-packages/torch/distributed/fsdp/fully_sharded_data_parallel.py:690: FutureWarning: FSDP.state_dict_type() and FSDP.set_state_dict_type() are being deprecated. Please use APIs, get_state_dict() and set_state_dict(), which can support different parallelisms, FSDP1, FSDP2, DDP. API doc: https://pytorch.org/docs/stable/distributed.checkpoint.html#torch.distributed.checkpoint.state_dict.get_state_dict .Tutorial: https://pytorch.org/tutorials/recipes/distributed_checkpoint_recipe.html .
(WorkerDict pid=2297616)   warnings.warn(

**Comment by hebiao064** [comment]
(WorkerDict pid=2297617)   warnings.warn(
Online DPO Training Progress:   3%|▎         | 3/87 [18:17<8:10:37, 350.44s/it, global_seqlen/min=66165, global_seqlen/max=71649, global_seqlen/minmax_diff=5484, global_seqlen/balanced_min=68768, global_seqlen/balanced_max=68768, global_seqlen/mean=68768.000, time/validation_run=141.534, val-core/DigitalLearningGmbH/MATH-lighteval/reward/mean@1=0.508,

**Comment by hebiao064** [comment]
Online DPO Training Progress:   9%|▉         | 8/87 [47:32<9:30:54, 433.60s/it, global_seqlen/min=60915, global_seqlen/max=66975, global_seqlen/minmax_diff=6060, global_seqlen/balanced_min=64730, global_seqlen/balanced_max=64731, global_seqlen/mean=64730.250, time/validation_run=138.318, val-core/DigitalLearningGmbH/MATH-lighteval/reward/mean@1=0.509

File: recipe/spin/test.sh [hunk 1]
```diff
@@ -2,15 +2,17 @@ set -e
...
-  data.val_files=$HOME/data/gsm8k/test.parquet \
+  data.train_files=/shared/user/bhe/data/verl/math/train.parquet \
+  data.val_files=/shared/user/bhe/data/verl/math/test.parquet \
   data.train_batch_size=256 \
   data.val_batch_size=1312 \
-  data.max_prompt_length=1024 \
+  data.max_prompt_length=2048 \
   data.max_response_length=1024 \
-  actor_rollout_ref.model.path=Qwen/Qwen2.5-3B-Instruct \
+  actor_rollout_ref.model.path=/shared/public/sharing/bhe/qwen/Qwen2-7B-Instruct/snapshots/f2826a00ceef68f0f2b946d945ecc0477ce4450c \
```
Comment (hebiao064) [inline recipe/spin/test.sh:15]: need attention on this model change

File: recipe/spin/online_dpo.sh [hunk 2]
```diff
@@ -13,12 +15,12 @@ export TOKENIZERS_PARALLELISM=false
...
 export WANDB_API_KEY=25c95cfb8dfe322ae6d944a369d2ae63b65d9ece
 export TORCH_MEMORY_SAVER=0
-MODEL_NAME="Qwen/Qwen2-7B-Instruct"
+MODEL_NAME="/shared/public/sharing/bhe/qwen/Qwen2-7B-Instruct/snapshots/f2826a00ceef68f0f2b946d945ecc0477ce4450c"
 BASE_MODEL_PATH=$MODEL_NAME
-REWARD_MODEL_PATH="trl-lib/Qwen2-0.5B-Reward"
+REWARD_MODEL_PATH="/shared/public/sharing/bhe/qwen/Qwen2-0.5B-Reward/snapshots/80fb188cabd3e854c2fb983f23a153c0e58a69e0"
 
-TRAIN_DATA_PATH="/root/data/gsm8k/train.parquet"
-VAL_DATA_PATH="/root/data/gsm8k/test.parquet"
```
Comment (hebiao064) [inline recipe/spin/online_dpo.sh:21]: need attention on the gsm change

**Comment by hebiao064** [comment]
Online DPO Training Progress:  29%|██▊       | 25/87 [2:48:34<7:20:02, 425.84s/it, global_seqlen/min=58153, global_seqlen/max=61990, global_seqlen/minmax_diff=3837, global_seqlen/balanced_min=60416, global_seqlen/balanced_max=60416, global_seqlen/mean=60416.000, time/validation_run=135.896, val-core/DigitalLearningGmbH/MATH-lighteval/reward/mean@1=0.539

**Comment by hebiao064** [comment]
image

Looks good @cedricbeta
